### PR TITLE
Enhancement/92 Bump WordPress and PHP minimums

### LINKED ISSUE #92: Bump WordPress and PHP minimums
Bump WP & PHP Minimums to the following
 
- [ ] WordPress to 5.7
- [ ] PHP to 7.4
- [ ] Remove any no-longer-needed conditional code for older version support.
- [ ] Take a look at adjusting our testing matrices, if applicable.
- [ ] Bump the minimums in plugin documentation and header fields.

### PR DESCRIPTION
**Pull request by faisal-alvi**
### Description of the Change
Bumped the minimum WordPress and PHP minimums.
* WordPress to 5.7
* PHP to 7.4

Closes #92

### How to test the Change
Check if the plugin is working as expected.
Check the Github Actions working without fails.

### Credits

Props @vikrampm1 @faisal-alvi @Sidsector9 

### Checklist:

- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [ ] All new and existing tests pass.

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -50,7 +50,7 @@ jobs:
     - name: Set PHP version
       uses: shivammathur/setup-php@v2
       with:
-        php-version: '7.2'
+        php-version: '7.4'
         coverage: none
         tools: prestissimo, composer:v2
```
Comment (Sidsector9) [inline .github/workflows/lint.yml]: `prestissimo` is unnecessary if you're using composer v2.

**Review by Sidsector9 — DISMISSED** [review]
@faisal-alvi all looks good except one minor change. Pre-approving the PR 👍

File: readme.txt [hunk 2]
```diff
@@ -44,10 +44,6 @@ Click the block inserter (`+` button) in the top left of the block editor, click
 
 Click the block inserter (`+` button) in the top left of the block editor, click the `Patterns` tab, select `Publisher Media Kit` in the dropdown, and select the specific Block Pattern that you want to add to your post/page.
 
-= Why do you require WordPress 5.5 and above?  Why not 5.0 and above? =
```
Comment (jeffpaul) [inline readme.txt:47]: should remove this from README.md as well

Comment (faisal-alvi) [reply readme.txt:47]: done https://github.com/10up/publisher-media-kit/pull/96/commits/8fd25d5046fff475c8080335a703d58406dd122e

File: .github/workflows/lint.yml (hunk 1 above)
Comment (faisal-alvi) [reply .github/workflows/lint.yml]: done https://github.com/10up/publisher-media-kit/pull/96/commits/4f0439cea82392c02c763a29455a297484101ce4

**Pull request merged**